### PR TITLE
partial merge of changes from Audacity 3.0.3

### PR DESCRIPTION
**Pull request by Be-ing**
Resolves: merging updates from Audacity

<details>
<summary>Checklist</summary>

- [x] I have signed off my commits using `-s` or `Signed-off-by`\* (See: [Contributing § DCO](https://github.com/tenacityteam/tenacity/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code\*
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving\*
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"\*

\* indicates required

</details>

### CODE REVIEW

**Review by nbsp — CHANGES_REQUESTED** [review]
You should sign off the commits: [DCO § B](https://developercertificate.org/) allows it, and it shifts the blame to Audacity if anything goes wrong.

Also, is there a way to get rid of the merge commits? That's a lot of them.

**Comment by Be-ing** [comment]
> You should sign off the commits: DCO § B allows it

That would require starting this work all over. You cannot rebase a merge commit without refixing the conflicts.

> Also, is there a way to get rid of the merge commits? That's a lot of them.

The alternative would be cherry picking every individual commit, which I was practically doing anyway with merge commits so maybe it's not a terrible idea.

**Comment by n0toose** [comment]
> The alternative would be cherry picking every individual commit, which I was practically doing anyway with merge commits so maybe it's not a terrible idea.

I do this. It's a bit hardcore and requires a lot of trial and error, but it definitely helps with fixing merge conflicts easily and just running `git rebase HEAD~<number> --sign-off` works. It also keeps the commit history very clean.

File: scripts/ci/upload_debug_symbols.sh [hunk 1]
```diff
@@ -11,6 +11,6 @@ curl -sL https://sentry.io/get-cli/ | bash
 
 SYMBOLS=$(find debug | xargs)
 
-${INSTALL_DIR}/sentry-cli --auth-token ${SENTRY_AUTH_TOKEN} --url ${SENTRY_HOST} upload-dif \
+${INSTALL_DIR}/sentry-cli --auth-token ${SENTRY_AUTH_TOKEN} --url https://${SENTRY_HOST} upload-dif \
```
Comment (n0toose) [inline scripts/ci/upload_debug_symbols.sh]: no

Comment (n0toose) [reply scripts/ci/upload_debug_symbols.sh]: Sorry to be blunt about this in the most silly manner possible, but have we left something over there that we should get rid of? I can't find this script on our own tree either.

Comment (Be-ing) [reply scripts/ci/upload_debug_symbols.sh]: This file should just be deleted.

**Comment by Be-ing** [comment]
This should be done all over with cherry picking. I'm not particularly feeling like doing that myself at the moment.

**Comment by n0toose** [comment]
We've got time! :D

**Pull request closed without merging**